.github/workflows/ci.yml

```diff
@@ -0,0 +1,443 @@
+name: Build and test gem
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  cross-compile:
+    strategy:
+      matrix:
+        platform:
+          - "x64-mingw32"
+          - "x86-mingw32"
+          - "x64-mingw-ucrt"
+    name: Cross-compile gem
+    runs-on: ubuntu-latest
+    container:
+      image: "ghcr.io/rake-compiler/rake-compiler-dock-image:1.4.0-mri-${{ matrix.platform }}"
+    steps:
+      - uses: actions/checkout@v4
+      - run: git config --global --add safe.directory /__w/tiny_tds/tiny_tds # shrug
+      - name: Install gems
+        shell: bash
+        run: bundle install
+      - name: Write used versions into file
+        shell: bash
+        run: bundle exec rake ports:version_file[${{ matrix.platform }}]
+      - name: Cache ports
+        uses: actions/cache@v4
+        with:
+          path: ports
+          key: cross-compiled-${{ hashFiles('**/.ports_versions') }}
+          restore-keys: |
+            cross-compiled-${{ hashFiles('**/.ports_versions') }}
+            cross-compiled-v2-
+      - name: Build gem
+        shell: bash
+        run: bundle exec rake gem:for_platform[${{ matrix.platform }}]
+      - name: Move gems into separate directory before persisting
+        run: |
+          mkdir -p artifacts/gem
+          mv pkg/*.gem artifacts/gem
+      - uses: actions/upload-artifact@v4
+        with:
+          name: gem-${{ matrix.platform }}
+          path: artifacts/gem
+
+  install-windows-mingw:
+    needs:
+      - cross-compile
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version:
+          - 2.7
+          - '3.0'
+
+    name: Install on Windows (MingW)
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+
+      - name: Download precompiled gem
+        uses: actions/download-artifact@v4
+        with:
+          name: gem-x64-mingw32
+
+      - name: Install native gem and test if TinyTDS loads
+        shell: pwsh
+        run: |
+          $rubyArchitecture = (ruby -e "puts RbConfig::CONFIG['arch']").Trim()
+          $gemVersion = (Get-Content VERSION).Trim()
+          $gemToInstall = "./tiny_tds-$gemVersion-$rubyArchitecture.gem"
+          
+          Write-Host "Looking to install $gemToInstall"
+          gem install --local "$gemToInstall"
+          
+          ruby -e "require 'tiny_tds'; puts TinyTds::Gem.root_path"
+          exit $LASTEXITCODE
+
+  test-windows-mingw:
+    needs:
+      - cross-compile
+    strategy:
+      fail-fast: false
+      matrix:
+        mssql-version:
+          #- 2017
+          - 2019
+          - 2022
+        ruby-version:
+          - 2.7
+          - '3.0'
+
+    name: Test on Windows (MingW)
```

Comment:
I think for PR check clarity, we'll benefit from seeing the ruby version (and maybe sql server version) directly in the job name. Probably same comment for all jobs as applicable. Also, I'm big fan of consistency across jobs, so maybe something like `test-$os (ruby $ruby_version)`, `build- ...`, etc. Looking at the list here: https://github.com/andyundso/tiny_tds/actions/runs/8806805186/job/24172426029

Reply:
I added a suggestion with 0b125119.